### PR TITLE
fix(engine): nullable

### PR DESCRIPTION
**Pull request by georgettica**
this one is more tricky as there are keyvaluepairs in here
I might have missed one or two

same as #13793

related to #13791

### CODE REVIEW

**Comment by georgettica** [comment]
Not surprising, this was a massive change..

Planning to fix with visual studio and hopefully will work much better

File: src/System.Management.Automation/engine/CmdletInfo.cs [hunk 1]
```diff
@@ -496,8 +496,8 @@ internal static string GetFullName(PSObject psObject)
                 // Handle the case in one or both of the properties might not be defined.
                 PSPropertyInfo nameProperty = psObject.Properties["Name"];
                 PSPropertyInfo psSnapInProperty = psObject.Properties["PSSnapIn"];
-                string nameString = nameProperty == null ? string.Empty : (string)nameProperty.Value;
-                string psSnapInString = psSnapInProperty == null ? string.Empty : (string)psSnapInProperty.Value;
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/CmdletInfo.cs:500]: It is not correct. See my comment in another PR.

File: src/System.Management.Automation/engine/CmdletParameterBinderController.cs [hunk 2]
```diff
@@ -539,7 +539,7 @@ private PSObject WrapBindingState()
                     }
                 }
 
-                result.Add(param, pair.Value);
+                result.Add(param, pair);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/CmdletParameterBinderController.cs:542]: It is not nullable.

**Review by iSazonov — COMMENTED** [review]
@georgettica Too many errors. Please fix based on my comments.

**Comment by georgettica** [comment]
will remove this PR aswell as I think some stuff will be better suited by others

**Pull request closed without merging**